### ISSUE TITLE
azure_rm_roleassignment_info:  when 'assignee' i set, role_definition_id filter not work

### ISSUE THREAD

**Issue by Eugen1987** [ansible/ansible_collections_azure#505]
##### SUMMARY

The module azure_rm_roleassignment_info does not return 'roleassignments' attribute even if the SP has assigned roles. 
Only happens when 'assignee' is set:
```yaml
    - name: Get resource roleassignmentinfo in scope
      azure_rm_roleassignment_info:
        scope: "{{ resource_id }}"
        assignee: "{{ sp_id }}"
        role_definition_id: "{{ roledefinition_id }}"
```
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure_rm_roleassignment_info

##### ANSIBLE VERSION
ansible 2.10.6
  config file = /Users/test/workplace/dap-platform-ops/ansible.cfg
  configured module search path = ['/Users/test/workplace/dap-platform-ops/library']
  ansible python module location = /Users/test/workplace/demo/lib/python3.8/site-packages/ansible
  executable location = /Users/test/workplace/demo/bin/ansible
  python version = 3.8.1 (v3.8.1:1b293b6006, Dec 18 2019, 14:08:53) [Clang 6.0 (clang-600.0.57)]

##### OS / ENVIRONMENT
```
Darwin topi 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar 4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
```
##### STEPS TO REPRODUCE

Create a playbook that tries to get the roles of an SP on a given azure component.

```yaml
- name: Get resource roleassignmentinfo in scope
    azure_rm_roleassignment_info:
    scope: "{{ resource_id }}"
    assignee: "{{ sp_id }}"
    role_definition_id: "{{ roledefinition_id }}"

```

##### EXPECTED RESULTS

```
roleassignments returns an array:
[{
    "id":"xxx",
    "name":"xxx",
    "principal_id":"xxx",
    "principal_type":"Group",
    "role_definition_id":"xxx",
    "scope":"xxx",
    "type":"Microsoft.Authorization/roleAssignments"
}]
```
##### ACTUAL RESULTS

```
roleassignments field is empty
```

**Comment by paultaiton** [comment]
possibly fixed in #464 

@Eugen1987 
I'm not completely understanding what you're reporting. You provide scope, assignee, role_definition_id and it doesn't return anything? Or that works but omitting one of the parameters causes it to fail?

**Comment by Eugen1987** [comment]
hi.
I have checked the Pr you indicated and it solves the problem.
Basically the problem is that when using the 'assignee' attribute and then filtering by 'role_definition_id' nothing is returned. It's a code problem:
![image](https://user-images.githubusercontent.com/19280657/115508553-bbc82580-a27d-11eb-8efc-64e72f09c49a.png)

when 'role_definition_id' is defined the results will always be empty.

**Comment by Fred-sun** [comment]
fixed in #464
